### PR TITLE
complete kubenetes port

### PR DESCRIPTION
**Pull request by ugoenyioha**

### CODE REVIEW

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/CustomersServiceClient.java [hunk 1]
```diff
@@ -30,6 +30,6 @@
     private final RestTemplate loadBalancedRestTemplate;
 
     public OwnerDetails getOwner(final int ownerId) {
-        return loadBalancedRestTemplate.getForObject("http://customers-service/owners/{ownerId}", OwnerDetails.class, ownerId);
+        return loadBalancedRestTemplate.getForObject("http://customers-service:8081/owners/{ownerId}", OwnerDetails.class, ownerId);
```
Comment (mszarlinski) [inline spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/CustomersServiceClient.java]: We don’t want to hardcode ports

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java [hunk 2]
```diff
@@ -40,4 +46,31 @@ public static void main(String[] args) {
     RestTemplate loadBalancedRestTemplate() {
         return new RestTemplate();
     }
+
+    @Bean
+    public ZuulFilter zuulFilter(){
```
Comment (mszarlinski) [inline spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java]: Please use more descriptive names. What is a purpose of this filter? Isn’t it some kind of debug?

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java [hunk 3]
```diff
@@ -15,20 +15,26 @@
...
 
+import java.util.List;
+
 /**
  * @author Maciej Szarlinski
  */
 @EnableZuulProxy
 @EnableDiscoveryClient
 @SpringBootApplication
+@EnableAutoConfiguration
```
Comment (mszarlinski) [inline spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java]: I think it is redundant due to use of ‘@SpringBootApplication’

File: spring-petclinic-api-gateway/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+zuul.debug.request=true
+logging.level.org.springframework=DEBUG
```
Comment (mszarlinski) [inline spring-petclinic-api-gateway/src/main/resources/application.properties]: We don’t want to enable debug mode by default. It should be switched on explicitly for debugging purposes.

File: spring-petclinic-vets-service/src/main/java/org/springframework/samples/petclinic/vets/web/VetResource.java [hunk 5]
```diff
@@ -19,6 +19,8 @@
 
 import java.util.List;
 
+import lombok.extern.slf4j.Slf4j;
```
Comment (mszarlinski) [inline spring-petclinic-vets-service/src/main/java/org/springframework/samples/petclinic/vets/web/VetResource.java]: Unused imports?

File: spring-petclinic-customers-service/src/test/java/org/springframework/samples/petclinic/customers/web/PetResourceDatabaseTest.java [hunk 6]
```diff
@@ -37,6 +39,9 @@
     @Autowired
     OwnerRepository ownerRepository;
 
+    @Autowired
+    private DiscoveryClient discoveryClient;
```
Comment (mszarlinski) [inline spring-petclinic-customers-service/src/test/java/org/springframework/samples/petclinic/customers/web/PetResourceDatabaseTest.java]: Is this bean used somewhere?

File: spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/CustomersServiceApplication.java [hunk 7]
```diff
@@ -18,6 +18,8 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
+import org.springframework.cloud.kubernetes.discovery.KubernetesDiscoveryClient;
```
Comment (mszarlinski) [inline spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/CustomersServiceApplication.java]: Unused imports?

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/VisitsServiceClient.java [hunk 8]
```diff
@@ -41,7 +41,7 @@
         final ParameterizedTypeReference<List<VisitDetails>> responseType = new ParameterizedTypeReference<List<VisitDetails>>() {
         };
         return petIds.parallelStream()
-            .flatMap(petId -> loadBalancedRestTemplate.exchange("http://visits-service/owners/{ownerId}/pets/{petId}/visits", HttpMethod.GET, null,
+            .flatMap(petId -> loadBalancedRestTemplate.exchange("http://visits-service:8082/owners/{ownerId}/pets/{petId}/visits", HttpMethod.GET, null,
```
Comment (arey) [inline spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/application/VisitsServiceClient.java]: Another hard-coded port

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java (hunk 3 above)
Comment (ugoenyioha) [reply spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java]: Sorry for being late to the comment party. ;)
Yeah i agree separate branch.

File: spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java (hunk 2 above)
Comment (ugoenyioha) [reply spring-petclinic-api-gateway/src/main/java/org/springframework/samples/petclinic/api/ApiGatewayApplication.java]: No use reminant from lots of debugging to get this to work. Will take it out.

**Pull request closed without merging**